### ISSUE TITLE
Failed with Jetson NX using tensorrt model and docker from nvcr.io/nvidia/tritonserver:22.02-py3

### ISSUE THREAD

**Issue by leemengwei** [triton-inference-server/server#4050]
root@4d79eebce660:/opt/tritonserver# ./bin/tritonserver --model-repository=/models/ --strict-model-config=false  --backend-directory=./backends/
WARNING: [Torch-TensorRT] - Unable to read CUDA capable devices. Return status: 999
I0313 13:39:09.358429 60 libtorch.cc:1306] TRITONBACKEND_Initialize: pytorch
I0313 13:39:09.358685 60 libtorch.cc:1316] Triton TRITONBACKEND API version: 1.8
I0313 13:39:09.358738 60 libtorch.cc:1322] 'pytorch' TRITONBACKEND API version: 1.8
2022-03-13 13:39:09.833051: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
2022-03-13 13:39:09.960356: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
I0313 13:39:09.960559 60 tensorflow.cc:2176] TRITONBACKEND_Initialize: tensorflow
I0313 13:39:09.960679 60 tensorflow.cc:2186] Triton TRITONBACKEND API version: 1.8
I0313 13:39:09.960741 60 tensorflow.cc:2192] 'tensorflow' TRITONBACKEND API version: 1.8
I0313 13:39:09.960777 60 tensorflow.cc:2216] backend configuration:
{}
I0313 13:39:09.965854 60 onnxruntime.cc:2319] TRITONBACKEND_Initialize: onnxruntime
I0313 13:39:09.965969 60 onnxruntime.cc:2329] Triton TRITONBACKEND API version: 1.8
I0313 13:39:09.966029 60 onnxruntime.cc:2335] 'onnxruntime' TRITONBACKEND API version: 1.8
I0313 13:39:09.966106 60 onnxruntime.cc:2365] backend configuration:
{}
W0313 13:39:10.030730 60 pinned_memory_manager.cc:236] Unable to allocate pinned system memory, pinned memory pool will not be available: unknown error
E0313 13:39:10.030836 60 server.cc:201] Failed to initialize CUDA memory manager: unable to get number of CUDA devices: unknown error
W0313 13:39:10.030894 60 server.cc:208] failed to enable peer access for some device pairs
E0313 13:39:10.031479 60 model_repository_manager.cc:1844] Poll failed for model directory 'nongken_trt': unable to get number of CUDA devices: unknown error
I0313 13:39:10.031677 60 server.cc:522] 
+------------------+------+
| Repository Agent | Path |
+------------------+------+
+------------------+------+

I0313 13:39:10.031841 60 server.cc:549] 
+-------------+-------------------------------------------------+--------+
| Backend     | Path                                            | Config |
+-------------+-------------------------------------------------+--------+
| pytorch     | ./backends/pytorch/libtriton_pytorch.so         | {}     |
| tensorflow  | ./backends/tensorflow1/libtriton_tensorflow1.so | {}     |
| onnxruntime | ./backends/onnxruntime/libtriton_onnxruntime.so | {}     |
+-------------+-------------------------------------------------+--------+

I0313 13:39:10.031926 60 server.cc:592] 
+-------+---------+--------+
| Model | Version | Status |
+-------+---------+--------+
+-------+---------+--------+

I0313 13:39:10.032549 60 tritonserver.cc:1932] 
+----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------+
| Option                           | Value                                                                                                                                            |
+----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------+
| server_id                        | triton                                                                                                                                           |
| server_version                   | 2.19.0                                                                                                                                           |
| server_extensions                | classification sequence model_repository model_repository(unload_dependents) schedule_policy model_configuration system_shared_memory cuda_share |
|                                  | d_memory binary_tensor_data statistics trace                                                                                                     |
| model_repository_path[0]         | /models/                                                                                                                                         |
| model_control_mode               | MODE_NONE                                                                                                                                        |
| strict_model_config              | 0                                                                                                                                                |
| rate_limit                       | OFF                                                                                                                                              |
| pinned_memory_pool_byte_size     | 268435456                                                                                                                                        |
| response_cache_byte_size         | 0                                                                                                                                                |
| min_supported_compute_capability | 6.0                                                                                                                                              |
| strict_readiness                 | 1                                                                                                                                                |
| exit_timeout                     | 30                                                                                                                                               |
+----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------+

I0313 13:39:10.032699 60 server.cc:252] Waiting for in-flight requests to complete.
I0313 13:39:10.032747 60 server.cc:267] Timeout 30: Found 0 live models and 0 in-flight non-inference requests
error: creating server: Internal - failed to load all models

Platform:
![image](https://user-images.githubusercontent.com/17986725/158062443-60b753c1-a10c-43f1-bdeb-8c765211f035.png)

**Comment by CoderHam** [comment]
@leemengwei there appears to be an error with the cuda environment. Which version of Triton are you using? Is it built by yourself or are you using the published tar release for jetson?

**Comment by CoderHam** [comment]
Please note that the Triton docker you pulled is likely for ARM SBSA. For jetson a triton docker is not supported and the way to use Triton on Jetson is using the tar file that is available in the release notes.

**Comment by CoderHam** [comment]
Yes, Triton has supported Jetson for a while. However it is supported not as a container but via a tar file. Release refer to the [release page](https://github.com/triton-inference-server/server/releases/tag/v2.19.0) and [docs/jetson.md](https://github.com/triton-inference-server/server/blob/r22.02/docs/jetson.md).

SBSA: Server Based System Architecture. Jetson devices have a different compute stack than traditional ARM SBSA.

Closing this ticket. Please use the intended tar file as detailed in the linked docs and open a new ticket if you run into any issues while doing to.

**Comment by leemengwei** [comment]
Problem solved. Very simple and compacted routine:

Step1, pull a docker from NGC:
docker pull nvcr.io/nvidia/tritonserver:22.02-py3

Step2, start this docker by:
docker run -it --gpus=all --rm --entrypoint /bin/bash -v/home/nvidia/user/triton_models/:/models/ triton219:v1

Step3, run server by: 
./bin/tritonserver --model-repository=/models/ --strict-model-config=false --backend-directory=./backends/

The only reason not worked yesterday is lack of "--gpus=all ", meanwhile, don't forget " --backend-directory=./backends/"

**Comment by CoderHam** [comment]
Interesting. The cuda and frameworks in the Triton container for ARM is built for Cuda 11.0. It is curious that it worked out of the box on a Cuda 10.2 machine.

@leemengwei thank you for brining this to our attention.

**Comment by leemengwei** [comment]
> Problem solved. Very simple and compacted guidance for the latercomes:
> 
> Step1, pull a docker from NGC: docker pull nvcr.io/nvidia/tritonserver:22.02-py3
> 
> Step2, start this docker by: docker run -it --gpus=all --rm --entrypoint /bin/bash -v/home/nvidia/user/triton_models/:/models/ nvcr.io/nvidia/tritonserver:22.02-py3
> 
> Step3, run server by: ./bin/tritonserver --model-repository=/models/ --strict-model-config=false --backend-directory=./backends/
> 
> The only reason not worked yesterday is lack of "--gpus=all ", meanwhile, don't forget " --backend-directory=./backends/"
> 
> Thus it seems triton on jetson IS supported by docker contianer, not only via tar file that requires mannual labor. @CoderHam

Sorry, for previous misleading. Must be too many host machines and dockers, got them muddled up....
nvcr.io/nvidia/tritonserver:22.02-py3 won't work,
Instead, use this Dockerfile:

```
From nvcr.io/nvidia/l4t-ml:r32.6.1-py3

RUN apt-get update && \
    apt-get install -y --no-install-recommends \
        software-properties-common \
        autoconf \
        automake \
        build-essential \
        cmake \
        git \
        libb64-dev \
        libre2-dev \
        libssl-dev \
        libtool \
        libboost-dev \
        libcurl4-openssl-dev \
        libopenblas-dev \
        rapidjson-dev \
        patchelf \
        zlib1g-dev

WORKDIR /tritonserver

RUN wget https://github.com/triton-inference-server/server/releases/download/v2.17.0/tritonserver2.17.0-jetpack4.6.tgz && \
    tar -xzf tritonserver2.17.0-jetpack4.6.tgz && \
    rm tritonserver2.17.0-jetpack4.6.tgz

ENTRYPOINT ["/tritonserver/bin/tritonserver"]
CMD ["--help"]
```

Build by：
docker build  -t triton217ml:v1 .

And run it by:
docker run -it --gpus=all --rm -v/home/nvidia/lmw/triton_models/:/models/ triton217ml:v1 --model-repository=/models/ --strict-model-config=false --backend-directory=./backends/
